dhis-2/dhis-api/src/main/java/org/hisp/dhis/expressiondimensionitem/ExpressionDimensionItemHelper.java

```diff
@@ -0,0 +1,112 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.expressiondimensionitem;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.hisp.dhis.category.CategoryOptionCombo;
+import org.hisp.dhis.common.BaseDimensionalItemObject;
+import org.hisp.dhis.common.DataDimensionItem;
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.dataelement.DataElement;
+import org.hisp.dhis.dataelement.DataElementOperand;
+
+/**
+ * Parsing the expression of ExpressionDimensionItem, provides collection of
+ * BaseDimensionalItemObjects.
+ */
+public class ExpressionDimensionItemHelper
+{
+    public static final Pattern pattern = Pattern
+        .compile( "[a-zA-Z0-9]{11}[.]?[a-zA-Z0-9]{0,11}[.]?[a-zA-Z0-9]{0,11}" );
+
+    private ExpressionDimensionItemHelper()
+    {
+        throw new UnsupportedOperationException( "helper" );
+    }
+
+    /**
+     * Provides collection of selected item types inside the expression
+     *
+     * @param manager {@link IdentifiableObjectManager} service for item
+     *        delivery
+     * @param dataDimensionItem {@link IdentifiableObjectManager} expression
+     *        dimension item
+     * @return collection of selected item types
+     */
+    public static List<BaseDimensionalItemObject> getExpressionItems( IdentifiableObjectManager manager,
+        DataDimensionItem dataDimensionItem )
+    {
+        if ( dataDimensionItem.getExpressionDimensionItem() == null )
+        {
+            return new ArrayList<>();
+        }
+
+        String expression = dataDimensionItem.getExpressionDimensionItem().getExpression();
+
+        List<String> expressionTokens = getExpressionTokens( pattern, expression );
+
+        List<BaseDimensionalItemObject> baseDimensionalItemObjects = new ArrayList<>();
+
+        expressionTokens.forEach( et -> {
+            String[] uids = et.split( Pattern.quote( "." ) );
+            if ( uids.length > 1 )
+            {
+                DataElementOperand deo = new DataElementOperand( manager.get( DataElement.class, uids[0] ),
+                    manager.get( CategoryOptionCombo.class, uids[1] ) );
+                baseDimensionalItemObjects.add( deo );
+            }
+            else if ( uids.length > 0 )
+            {
+                baseDimensionalItemObjects.add( manager.get( DataElement.class, uids[0] ) );
+            }
+        } );
+
+        return baseDimensionalItemObjects;
+    }
+
+    /**
+     * Expression parser
+     *
+     * @param expression
+     * @return collection of tokens
+     */
+    public static List<String> getExpressionTokens( Pattern pattern, String expression )
```

Comment:
We could improve this Javadoc description. Also, the parameter `pattern` is not mentioned.